### PR TITLE
xtask: added handy scripts for inserting and removing test data files

### PR DESCRIPTION
**Pull request by Veetaha**
Inserting or deleting test data files in the middle of the directory gets tedious.
You always have to bookkeep the tests numbering.
I added two scripts to automatically update test file names when inserting new test data files and removing them.
@matklad proposed to do this in Python, but... Hell no!

### CODE REVIEW

File: xtask/src/lib.rs [hunk 1]
```diff
@@ -106,6 +106,81 @@ fn install_clippy() -> Result<()> {
...
+    Ok(())
+}
+
+fn parse_test_path(test_path_relative_to_crates: &str) -> Result<(PathBuf, PathBuf, u32)> {
+    let test_file_path = project_root().join("crates").join(test_path_relative_to_crates);
+
+    let mut test_dir = test_file_path.clone();
+    let test_id = parse_test_id(test_file_path.file_name().unwrap().to_str().unwrap())?;
+
+    test_dir.pop();
```
Comment (Veetaha) [inline xtask/src/lib.rs]: Should swap this lines...

**Comment by matklad** [comment]
r=me

bors d+

**Comment by kjeremy** [comment]
Can we get some docs?

**Comment by lnicola** [comment]
Stupid question, but do the files need to be numbered?

**Comment by matklad** [comment]
Sort of: the idea is to create a test suite with progressive difficulty,
which should allow for easier alternative implementations. But that’s
probably not *that* valuable, so I am ok with just removing numbers

**Comment by VitaliiKryvenkoTR** [comment]
@kjeremy Sure, I will add `--help` option similar to how it is done in `cargo xtask install --help`

**Comment by Veetaha** [comment]
@kjeremy, sure I'll add `--help` option similar to how it is done with `cargo xtask install --help`

**Comment by Veetaha** [comment]
@lnicola , @matklad, @kjeremy, it seems to me that we have some misconception about test data files naming. Can we, please, elaborate on this question? Does one have something against prefixing tests with a 4-digit sequential id number? If you think we could use a better practice, please say so

**Comment by lnicola** [comment]
I asked because I've seen that kind of file numbering (e.g. to enforce the execution order of SQL scripts) and it was a pain to maintain, and had no real benefits. Note how there are 15 lexer tests with a four-digit prefix and one with a five-digit one, probably by mistake. As you've said:

> Inserting or deleting test data files in the middle of the directory gets tedious.

So I was just asking whether there's a benefit (or need) to having that prefix.

**Comment by Veetaha** [comment]
I think we should postpone merging this PR, while we have discussions on our test infrastructure. If we decide to leave numbered tests as is, then we should merge it, otherwise, let's not hurry and try to engineer what best fits our needs.
Discussions about the test infra are carried out in [this zulip thread](https://rust-lang.zulipchat.com/#narrow/stream/185405-t-compiler.2Fwg-rls-2.2E0/topic/Moving.20to.20external.20tests)

**Comment by matklad** [comment]
I like to leave a couple of notes here

* I don't have strong opinion here, I believe that this genuinely does not matter now, as the parser is more or less *done* for the time being.
* What would be a significant (but hard) improvement to the status quo is the "official" standardization of the syntax, by the combination of a formal grammar and a reference test suite. I think https://github.com/rust-lang/wg-grammar is supposed to do this, but the progress has been slow
* I feel the amount of discussion generated here is disproportional to the significance of the issue

**Comment by Veetaha** [comment]
Okay, since we didn't come to an ultimate conclusion on the question about numbering our tests, lets just vote.
* We keep sequentially numbering our test data files and merge these utility scripts.
* We no longer number our tests (remove number prefixes from test data files) and close this PR.

Reasonable?

**Comment by Veetaha** [comment]
@matklad, sorry for blowing an elephant out of a bug here. I see this won't go anywhere until we have a strong opinion on it. And let me represent one.

Whilst I have such a power, I'll close this PR and propose us to switch to non-numbered tests. This question is actual in the light of recent development of `ra_syntax` crate and the way we will do tree validation (which is a separate topic I would like to discuss one I have time).
The reason for not doing numbered tests is the inconvenience that it brings for the developers to add and change existing tests. This PR just fixes the symptoms of the inconvenience, I'd like to remove the root ...

Sorry if I am proposing to change already working stuff ...

**Comment by lnicola** [comment]
My feeling is that the file naming isn't such a big issue yet to warrant a change in either direction. At some point there might be more tests and the numbering might become painful enough.

**Pull request closed without merging**